### PR TITLE
Support chat history in PM chatboxes

### PR DESCRIPTION
**Pull request by urkerab**
kota mentioned this when I was writing PR #477; kota also asked for a deduplication feature, although not exactly the same as the one I've actually implemented.

I made the chat history object global and split some of the key handling code out to it to reduce duplication.

### CODE REVIEW

**Comment by kotarou3** [comment]
Personally I prefer just checking if the previous message was a duplicate, and I think that's what most chat history features in other programs do.

Other than that, lgtm

**Comment by Zarel** [comment]
I'm not sure global chat history is desirable. Can you poll people?

**Comment by Zarel** [comment]
I think the chat history object can be left in `client-chat.js`

**Comment by urkerab** [comment]
Sorry for the delay, but I misread your comment: this isn't global chat history; each room and PM creates their own chat history.

File: js/client-mainmenu.js [hunk 1]
```diff
@@ -295,6 +296,7 @@
 							$chat.append('<div class="chat">User ' + userid + ' no longer ignored.</div>');
 						}
 					} else {
+						this.chatHistories[userid].push(text);
```
Comment (Zarel) [inline js/client-mainmenu.js]: Why is this here? Shouldn't `/ignore` and `/unignore` be added to the histories?

Comment (urkerab) [inline js/client-mainmenu.js]: I actually added line 299 before I knew that line 282 existed, and then removed line 282 as part of a cleanup sweep without realising what I'd done. I've moved it back (so to speak).

**Comment by Zarel** [comment]
At some point, we should refactor this so the main menu actually contains objects representing each PM box.
